### ISSUE TITLE
Select Dropdown Behaviour

### ISSUE THREAD

**Issue by vishnup-ditto** [mantinedev/mantine#3296]
### What package has an issue

@mantine/core

### Describe the bug

While using the Select component, if we scroll in a state where the dropdown is open and flip the dropdown.Then search the Select Fields, the position of the dropdown moves away from the input, creating a visual bug. 

https://app.usebubbles.com/pFWDWrgKVFmHwG2fqD64ka/my-video-recording

^^ A recording of the issue in the mantine.dev examples.

### What version of @mantine/hooks page do you have in package.json?

v5.9.4

### If possible, please include a link to a codesandbox with the reproduced problem

Can be reproduced in the mantine.dev website examples.

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Not sure.

**Comment by jalalbmnf** [comment]
Hi @rtivital , can i work on this issue ?

**Comment by rtivital** [comment]
sure

**Comment by jalalbmnf** [comment]
@vishnup-ditto Do you working on this issue?

**Comment by vishnup-ditto** [comment]
No,I am not sure how to fix this. Is mantine using `@floating-ui` for dropdown placements?

**Comment by jalalbmnf** [comment]
I guess yes, `"@floating-ui/react-dom-interactions": "^0.10.1"`

Please, inform me the end of the day, if you will work, then I will choose another issue :)

**Comment by vishnup-ditto** [comment]
No I am not working on fixing this issue, please assign it to yourself and feel free to fix.

**Comment by jalalbmnf** [comment]
@rtivital can you assign this issue to me?

**Comment by rtivital** [comment]
You can work on this issue

**Comment by jalalbmnf** [comment]
I spent 1.5 days fixing this bug and logged everything but unfortunately, I did not find the solution. 😅

**Comment by dipiash** [comment]
I'm faced with this problem also. 

For temporary fix I'm using `positionDependencies` props on select and subscribe to `onSearchChange`:
```tsx
const [positionDependencies, setPositionDependencies] = useState<string | null>(null)

const handleSearch = useCallback(
  (query: string) => {
    setPositionDependencies(query)
    onSearchChange?.(query)
  },
  [onSearchChange],
)

<Select
  ...
  searchable
  positionDependencies={[positionDependencies]} 
  onSearchChange={handleSearch}
  ...
>
```

I'm trying to investigate more

**Comment by rtivital** [comment]
Fixed in 5.10.3
